lib/spack/llnl/util/filesystem.py

```diff
@@ -1389,7 +1389,15 @@ def find_libraries(libraries, root, shared=True, recursive=False):
     # List of libraries we are searching with suffixes
     libraries = ['{0}.{1}'.format(lib, suffix) for lib in libraries]
 
-    return LibraryList(find(root, libraries, recursive))
+    for subdir in ('lib', 'lib64'):
```

Comment:
I wonder if we should check `lib64` first. If a library exists in both `lib` and `lib64`, which one do we want to default to?

Reply:
I just copied the same search order as `spec.py::_default_libs_handler`

Reply:
`find_system_libraries` searches `<prefix>/lib64` first

Reply:
That's because I wrote `find_system_libraries` in #3777 and @v-dobrev wrote the faster `_default_libs_handler` implementation in #7553. @v-dobrev any opinions on searching `lib` or `lib64` first?

Reply:
I don't think I had a specific reason to put `lib` before `lib64`. Switching the order for consistency makes sense to me.

Reply:
I just think if there happens to be a library in both `lib64` and `lib`, the one in `lib` is more likely to be 32-bit, which a user would be less likely to want to build with.

Reply:
My understanding of [FHS](https://en.wikipedia.org/wiki/Filesystem_Hierarchy_Standard) is that `/usr/lib/` contains libraries needed by executables in `/usr/bin` - they can be 32 or 64 bit.  `/usr/lib<qual>` is optional.  For what it’s worth, on my RHEL machine, the blas and lapack libraries are in `/usr/lib64`.

Reply:
IMO the order can be kept the same for now